libraries/libfc/src/log/gelf_appender.cpp

```diff
@@ -117,9 +132,20 @@ namespace fc
 
       if (my->gelf_endpoint)
       {
-        my->gelf_socket.initialize(io_service);
+        my->gelf_socket.initialize(my->io_context);
         my->gelf_socket.open();
         std::cerr << "opened GELF socket to endpoint " << my->cfg.endpoint << "\n";
+
+        my->thread = std::thread([this] {
```

Comment:
I think this calls `terminate()` on `SIGHUP` which calls `initialize()`

Reply:
SIGHUP would eventually create a new logger object and won't call `initialize()` on the same object twice.